packages/playground/src/components/execution/key-value-table.tsx

```diff
@@ -25,13 +25,98 @@ interface KeyValueTableProps {
   tableSpec: KeyValuePair[]
 }
 
+//manage style for repeated use
+const accordionContentStyle = 'w-full pl-0 pr-0 border-0 pb-0'
+const specTitleStyle = 'text-ring flex-grow text-left -ml-1'
+
 export const KeyValueTable: React.FC<KeyValueTableProps> = ({
   className,
   tableTitleName,
   tableTitleVal,
   tableSpec
 }) => {
-  // ToDO:const getAccordian = () => <div></div> // do this function recursively
+  const renderListItems = (items: KeyValuePair[], level: number = 1) => {
+    //detect if the listItems is objects or array, tailwind css will not generate
+    const listItems = Array.isArray(items) ? items : [items]
+
+    return listItems.map((item, index: number) => {
+      if (typeof item.value === 'string') {
+        return (
+          <ul className="border-b" key={index}>
+            <li className="py-2.5 inline-block w-1/2 " style={{ paddingLeft: `${level * 1 + 2}rem` }}>
+              <Text size={2} weight="normal" className="text-ring">
+                {item.name}
+              </Text>
+            </li>
+            <li className="pr-2.5 py-2.5 inline-block w-1/2">
+              <Text size={2} weight="normal" className="text-ring">
+                {item.value}
+              </Text>
+            </li>
+          </ul>
+        )
+      } else if (Array.isArray(item.value) || typeof item.value === 'object') {
+        return (
+          <Accordion type="single" key={index} className="border-0" defaultValue={item.name} collapsible>
+            <AccordionItem value={item.name} className="border-0">
+              <AccordionTrigger
+                className="w-full pt-2 pb-2 pr-4 flex"
+                leftChevron
+                style={{ paddingLeft: `${level * 1 + 2}rem` }}>
+                <Text size={2} weight="normal" className={specTitleStyle}>
+                  {item.name}
+                </Text>
+              </AccordionTrigger>
+              <AccordionContent className={accordionContentStyle}>
+                {renderListItems(item.value, level + 1)}
+              </AccordionContent>
+            </AccordionItem>
+          </Accordion>
+        )
+      }
+      return null
+    })
+  }
+
+  const renderTableRows = (tableSpec: KeyValuePair[]) => {
+    return tableSpec.map((item, index: number) => {
+      if (typeof item.value === 'string') {
+        return (
+          <TableRow key={index} className="border-b">
```

Comment:
Would adding a variant to the Table component to handle the styling delta be better than lots of inline alterations?